x-pack/legacy/plugins/apm/public/components/shared/charts/TransactionCharts/ChoroplethMap/Legend.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import styled from 'styled-components';
+import { fontSizes } from '../../../../../style/variables';
+import { asTime } from '../../../../../utils/formatters';
+
+const LegendLabel = styled.span`
+  font-size: ${fontSizes.small};
+`;
+
+export const Legend: React.SFC<{
+  getColorStyle: (scale: number) => string;
+  min: number;
+  max: number;
+}> = ({ getColorStyle, min, max }) => (
+  <div>
+    <div
+      style={{
+        height: 6,
+        opacity: 0.75,
+        backgroundImage: `linear-gradient(to right, ${[0, 0.5, 1]
+          .map(getColorStyle)
+          .join(',')})`
+      }}
+    />
+    <div
+      style={{
+        display: 'flex',
+        justifyContent: 'space-between'
+      }}
+    >
+      <LegendLabel>{asTime(min)}</LegendLabel>
+      <LegendLabel>{asTime(max)}</LegendLabel>
```

Comment:
When using `asTime` you use a new formatter every time. So if `min` could be formatted as milliseconds while `max` is formatted as seconds. Often you want to use the same formatter throughout a component so units don't change. In that case I'd probably base the formatter on the `max` value.

You can do this:

```ts
const formatter = getTimeFormatter(max)

// ---

<LegendLabel>{formatter(min)}</LegendLabel>
<LegendLabel>{formatter(max)}</LegendLabel>
```